### PR TITLE
Add night mode with Material UI

### PR DESCRIPTION
**Pull request by forabi**
* [x] ~Figure out how to apply illumination effects with variable colors~ (not needed any more because we replaced illuminations with pulse effect)
* [x] ~Hide toggle for browsers that do not support CSS variables~
* [x] Persist selected theme across page reloads
* [x] Log night mode clicks
* [x] ~Test how changes look on Chrome 41 (which does not support CSS Custom Properties)~
* [x] Figure out how to handle Facebook comments
* [x] Test color contrast
* [x] Algolia Logo
* [x] NP name
* [x] Sources List
* [x] Spinner, links
* [x] Night mode toggle
* [x] Facebook logo and menu item text
* [x] Home page search field
* [x] Shadows (NP photo)
* [x] Dialogs
* [x] Hide log in menu item

### CODE REVIEW

**Comment by forabi** [comment]
Changes are available on https://hollowverse.com/Tom_Hanks?branch=night-mode-material

@wholesomedev Let me know if I missed something. MUI already has a "dark" option when creating the theme which took care of most things but I had to wrap some text elements with `Typography` and adjust images and FB comments to make them aware of night mode. I didn't change the dark colors set by MUI for the background, paper and text. I noticed most apps which have a night mode on Android use these colors. You can compare the changes with the previous night mode implementation on https://hollowverse.com/Tom_Hanks?branch=night-mode (one thing you will immediately notice is that MUI components look more professional :smile:)

**Comment by wholesomedev** [comment]
Yep, the new branch looks great!

I did notice that the MUI components look more professional! And my suggestion to make the card darker in the old implementation certainly didn't help 😅

**Comment by forabi** [comment]
:tada:

File: src/app/__tests__/authInit.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { createTestContext, TestContext } from 'helpers/testHelpers';
 import { fireEvent } from 'react-testing-library';
 
-describe('authentication', () => {
+// eslint-disable-next-line jest/no-disabled-tests
```
Comment (wholesomedev) [inline src/app/__tests__/authInit.ts:4]: Should we add a comment about why these tests are skipped?

File: src/app/components/NavBarInput/NavBarInput.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export const NavBarInput = withStyles(styles)(
+  class extends React.Component<Props> {
+    render() {
+      const { classes, ...rest } = this.props;
+
+      return <DefaultInput disableUnderline classes={classes} {...rest} />;
+    }
+  },
+);
```
Comment (wholesomedev) [inline src/app/components/NavBarInput/NavBarInput.tsx:35]: 👍

File: src/app/store/features/theme/reducer.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    props: {
+      MuiTooltip: {
+        enterDelay: 400,
+      },
+      MuiPaper: {
+        elevation: 1,
+      },
+    },
+  }),
+);
```
Comment (wholesomedev) [inline src/app/store/features/theme/reducer.ts:103]: Interesting that the theme is in the state, but I guess it makes sense

**Review by wholesomedev — COMMENTED** [review]
LGTM! 🌠

**Pull request merged**